### DISCUSSION TITLE
Trimming clusters not connected from one side, through the entire domain, to another

### DISCUSSION THREAD

**Discussion by andtoni** [PMEAL/OpenPNM#2925] [Q&A] [answered]
Hi all,

Firstly, thank you for creating such an amazing module - I am in need of some advice.

I have an extracted and labelled network. I am wanting to identify clusters of pores and throats that are continually connected from one side of a domain, say the 'left',  all the way through the thickness of the network, to another domain, say the 'right', i.e. through the scaffold in an x direction, or y or z.

I would then like to trim all other pore clusters that are not connected through the entire domain in the specified direction.

I note that in the percolation example, csg.connected_components from csgraph is almost used to this affect, however, I cannot figure out how I may apply this to a network object.

Does anyone by chance have any advice around this? I would sincerely appreciate it.

**Comment by jgostick** [comment, marked as answer]
Hi @andtoni
We actually have a function which does exactly this: `remove_isolated_clusters`.  Just use `inlets` as the boolean sum of both inlet and outlet faces.  

Also, maybe this example will be helpful: https://openpnm.org/examples/reference/networks/managing_clustered_networks.html
